GVFS/GVFS.Common/GVFSEnlistment.cs

```diff
@@ -104,13 +104,13 @@ public static GVFSEnlistment CreateFromDirectory(string directory, string gitBin
                 string enlistmentRoot;
                 if (!GVFSPlatform.Instance.TryGetGVFSEnlistmentRoot(directory, out enlistmentRoot, out errorMessage))
                 {
-                    return null;
+                    throw new InvalidRepoException($"Could not get enlistment root. Error: {errorMessage}");

```

Comment:
Does `CreateWithoutRepoUrlFromDirectory` above need to follow this pattern as well?  Maybe we could have one method since they are both so similar?

Reply:
That's a good idea, let me know what you think of the changes/approach in the next commit